administrator/components/com_associations/models/fields/associatedcomponent.php

```diff
@@ -64,40 +64,46 @@ protected function getGroups()
 			'com_users',
 		);
 
+		$user       = JFactory::getUser();
+		$userId     = $user->id;
+
 		// Get all admin components.
 		foreach (glob(JPATH_ADMINISTRATOR . '/components/*', GLOB_NOSORT | GLOB_ONLYDIR) as $componentAdminPath)
 		{
 			$component           = basename($componentAdminPath);
 			$componentModelsPath = $componentAdminPath . '/models';
 
-			// Only components that exist also in the site client, aren't in the excluded components array and have models.
-			if (!is_dir($componentModelsPath) || in_array($component, $excludeComponents))
-			{
-				continue;
-			}
-
-			// Check if component uses associations, by checking is models.
-			foreach (glob($componentModelsPath . '/*.php', GLOB_NOSORT) as $modelFile)
+			if ($user->authorise('core.manage', $component . $this->referenceId))
 			{
```

Comment:
thsi works, but it's not engough

try the following:
- Create a user assigned to "Administrator" group
- Go to com_newsfeeds and say the Administrator groups is Denied in "Access Administration Interface"
- Login with that user in a new private window or other broewser
- Check the components menu news feeds is not there.
- Go to "Multilingual Associations" and the check newsfeed is not in the components list
- All ok until now...
- Now, access /administrator/index.php?option=com_associations&view=associations&component=com_newsfeeds.newsfeed&language=en-GB
 Voliá **ACL breach**...

So you need to make that check also here  (https://github.com/joomla-projects/gsoc16_improved-multi-lingual/blob/staging/administrator/components/com_associations/views/associations/view.html.php#L81) and also you will need to make it in the side by side view.